Encapsulation/Elena/ComplexNumber.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Encapsulation.Elena
+{
+    class ComplexNumber
+    {
+        private float real;
+        private float imegion;
+
+        public ComplexNumber()
+        {
+            real=0;
+            imegion=0;
+        }
+
+        public ComplexNumber(float r, float i)
+        {
+            real=r;
+            imegion=i;
+        }
+
+        public float Real
+        {
+            get { return real; }
+        }
+
+        public float Imegion
+        {
+            get { return imegion; }
+        }
+
+        public static ComplexNumber operator+(ComplexNumber operand1, ComplexNumber operand2)
+        {
+            return new ComplexNumber(operand1.real + operand2.real, operand1.imegion + operand2.imegion);
+        }
+
+        public static ComplexNumber operator -(ComplexNumber operand1, ComplexNumber operand2)
+        {
+            return new ComplexNumber(operand1.real - operand2.real, operand1.imegion - operand2.imegion);
+        }
+
+        public static ComplexNumber operator *(ComplexNumber operand1, int multiplier)
+        {
+            return new ComplexNumber(operand1.real *multiplier, operand1.imegion*multiplier);
+        }
+
+        public static ComplexNumber operator *(ComplexNumber operand1, ComplexNumber operand2)
+        {
+            return new ComplexNumber((operand1.real * operand2.real-operand1.imegion*operand2.imegion), (operand1.imegion * operand2.real+operand1.imegion*operand2.imegion));
+        }
+
+        public static ComplexNumber operator /(ComplexNumber operand1, ComplexNumber operand2)
+        {
+            float real;
+            float imegion;
+
+            real = (operand1.real * operand2.real + operand1.imegion * operand2.imegion) / (operand2.imegion * operand2.imegion + operand2.real * operand2.real);
+            imegion=(operand1.imegion*operand2.real-operand1.real*operand2.imegion)/ (operand2.imegion * operand2.imegion + operand2.real * operand2.real);
+            return new ComplexNumber(real, imegion);
+        }
+
+        public static ComplexNumber operator /(ComplexNumber operand1, int div)
+        {
+            if (div != 0)
+            {
+                return new ComplexNumber(operand1.real / div, operand1.imegion / div);
+            }
+            else { return null; }
```

Comment:
it is bad habit to return null. Return some empty object or (better in this case) throw something like DivisionByZeroException